archunit-tooling-test/build.gradle

```diff
@@ -0,0 +1,144 @@
+    implementation dependency.junitPlatformLauncher
+    implementation dependency.guava
+    implementation dependency.systemLambda
+    implementation dependency.mavenInvoker
+    implementation dependency.surefireReportParser
+    implementation dependency.commonsIo
+    implementation dependency.gradleToolingApi
+    implementation dependency.surefireSharedUtils
+
+    compileOnly dependency.findBugsAnnotations
+
+    testFixturesImplementation project(path: ':archunit')
+    testFixturesImplementation project(path: ':archunit-junit5-engine-api')
+    testFixturesImplementation project(path: ':archunit-junit5-api')
+    testFixturesImplementation project(path: ':archunit-junit5-engine')
+    testFixturesImplementation project(path: ':archunit-junit4')
+    testFixturesImplementation dependency.junitPlatform
+    testFixturesImplementation dependency.junit5JupiterEngine
+    testFixturesImplementation dependency.junit5JupiterApi
+
+    testImplementation dependency.assertj
+    testImplementation dependency.junitJupiterParams
+    testImplementation dependency.junitPioneer
+}
+
+String getMavenRepoPath(Project project) {
+    def pubs = project.publishing.publications.findAll() as Set<MavenPublication>
+    def pub = pubs[0]
+    if (!pub){
+        return null
+    }
+    String groupPath = pub.groupId.replace('.', '/')
+    return "$groupPath/$pub.artifactId"
+}
+
+def resolveAgainstMavenLocal(String path) {
+    new File(new File(repositories.mavenLocal().url), path)
+}
+
+def deleteSafely(File file) {
+    if (file.exists()) {
+        println "Deleting ${file.toURI().toString()}"
+        delete file
+    } else {
+        println "Could not find ${file.toURI().toString()}"
+    }
+}
+
+task removeFromMavenLocal {
+    group = 'other'
+    description = 'Remove artifact and all versions/classifiers/flavors from local maven repository.'
+    doLast {
+        [project(':archunit-junit5-engine-api'),
+         project(':archunit-junit5-engine'),
+         project(':archunit-junit5-api'),
+         project(':archunit-junit5'),
+         project(':archunit-junit4'),
+         project(':archunit')].each { project ->
+            File file = resolveAgainstMavenLocal(getMavenRepoPath(project))
+            deleteSafely file
+        }
+    }
+}
+
+task publishJUnitDepsToMavenLocal {
+    group = 'other'
+    description = 'Publishes ArchUnit JUnit artifacts to local Maven repo'
+    dependsOn ':archunit-junit5-engine-api:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5-engine:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5-api:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit5:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit-junit4:publishMavenJavaPublicationToMavenLocal'
+    dependsOn ':archunit:publishMavenJavaPublicationToMavenLocal'
+}
+
+gradle.taskGraph.whenReady { graph ->
+    if (graph.hasTask(publishJUnitDepsToMavenLocal)) {
```

Comment:
Needed because all the `publishMavenJavaPublicationToMavenLocal` tasks depend on `test` in their respective subprojects, so basically, `publishJUnitDepsToMavenLocal` would re-trigger the tests for all of those subprojects that we need in the local Maven repo.

Perhaps there's a better way of doing the same, didn't really investigate